Content.Shared/Temperature/Components/TemperatureComponent.cs

```diff
@@ -6,19 +6,42 @@ namespace Content.Shared.Temperature.Components;
 /// <summary>
 /// Handles tracking the temperature and heat capacity of an entity.
 /// </summary>
+[Access(typeof(SharedTemperatureSystem))]
 [RegisterComponent]
 public sealed partial class TemperatureComponent : Component
 {
     /// <summary>
     /// Surface temperature which is modified by the environment.
     /// </summary>
-    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    [DataField] // VV handled through SharedTemperatureSystem
     public float CurrentTemperature = Atmospherics.T20C;
 
+    /// <summary>
+    /// The amount of energy (in joules) required to increase the temperature of this entity by 1 kelvin.
+    /// </summary>
+    /// <remarks>
+    /// Should not be directly accessed. Use <see cref="SharedTemperatureSystem.GetHeatCapacity"/>.
+    /// </remarks>
+    [Access(typeof(SharedTemperatureSystem), Other = AccessPermissions.None)]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public float CachedHeatCapacity = float.NaN;
```

Comment:
Setting this to NaN could possibly lead to nasty NaN propagation in case that something is wrong with the dirtying logic.